### PR TITLE
[ALLUXIO-2838] Add delete shell with alluxioOnly Option

### PR DESCRIPTION
**Pull request by Jason0812**
use delete -alluxioOnly to delete metadata and block data just in alluxio space, and not to delete in ufs space if ufs mounted
Addresses: https://alluxio.atlassian.net/browse/ALLUXIO-2838

### CODE REVIEW

**Comment by Jason0812** [comment]
Address: https://alluxio.atlassian.net/browse/ALLUXIO-2838

**Comment by Jason0812** [comment]
alluxio-bot, check this please

**Comment by Jason0812** [comment]
alluxio-bot, check this please

**Comment by Jason0812** [comment]
Jenkins, test this please

**Comment by Jason0812** [comment]
Jenkins, test this please

**Comment by Jason0812** [comment]
Jenkins, test this please

**Comment by Jason0812** [comment]
Jenkins, test this please

**Comment by uronce-cc** [comment]
@Jason0812 Any update?

File: shell/src/main/java/alluxio/shell/command/RmCommand.java [hunk 1]
```diff
@@ -65,21 +64,26 @@ protected void runCommand(AlluxioURI path, CommandLine cl) throws AlluxioExcepti
           path.getPath() + " is a directory, to remove it, please use \"rm -R <path>\"");
     }
 
-    DeleteOptions options = DeleteOptions.defaults().setRecursive(recursive);
-    if (cl.hasOption(REMOVE_UNCHECKED_OPTION_CHAR)) {
-      options.setUnchecked(true);
-    }
+    boolean isAlluxioOnly = cl.hasOption("alluxioOnly");
+    //System.out.println("isAlluxioOnly: " + isAlluxioOnly);
```
Comment (uronce-cc) [inline shell/src/main/java/alluxio/shell/command/RmCommand.java]: remove this line.

File: shell/src/main/java/alluxio/shell/command/RmCommand.java [hunk 2]
```diff
@@ -65,21 +64,26 @@ protected void runCommand(AlluxioURI path, CommandLine cl) throws AlluxioExcepti
...
+    DeleteOptions options = DeleteOptions.defaults()
+        .setRecursive(recursive).setAlluxioOnly(isAlluxioOnly);
     mFileSystem.delete(path, options);
-    System.out.println(path + " has been removed");
+    if (!isAlluxioOnly) {
+      System.out.println(path + " has been removed");
+    } else {
+      System.out.println(path + " just been removed in Alluxio Space, but Ufs space still exists");
```
Comment (uronce-cc) [inline shell/src/main/java/alluxio/shell/command/RmCommand.java]: path + " has been removed from Alluxio space, but still exists in the under storage."

File: shell/src/main/java/alluxio/shell/command/RmCommand.java [hunk 3]
```diff
@@ -65,21 +64,26 @@ protected void runCommand(AlluxioURI path, CommandLine cl) throws AlluxioExcepti
...
   public String getUsage() {
-    return "rm [-R] <path>";
+    return "rm [-R] [-alluxioOnly] <path>";
   }
 
   @Override
   public String getDescription() {
-    return "Removes the specified file. Specify -R to remove file or directory recursively.";
+    return "Removes the specified file. Specify -R to remove file or directory recursively."
+        + "Specify -alluxioOnly just remove block data and metadata in Alluxio space";
```
Comment (uronce-cc) [inline shell/src/main/java/alluxio/shell/command/RmCommand.java]: " Specify -alluxioOnly to remove data and metadata from Alluxio space, but the data in under storage is not deleted."

**Comment by yupeng9** [comment]
@Jason0812 any updates on this PR?

**Comment by uronce-cc** [comment]
@Jason0812 any update?

**Comment by maobaolong** [comment]
@Jason0812  this is a useful option, any other updates?  I see you remove the unchecked option, Why?

**Comment by maobaolong** [comment]
@Jason0812 @yupeng9 @uronce-cc We should modify the DefaultFileSystemMaster.delete() to:

```java
public void delete(AlluxioURI path, DeleteOptions options) throws IOException,
      FileDoesNotExistException, DirectoryNotEmptyException, InvalidPathException,
      AccessControlException {
    Metrics.DELETE_PATHS_OPS.inc();
    try (JournalContext journalContext = createJournalContext();
         LockedInodePath inodePath = mInodeTree.lockFullInodePath(path, InodeTree.LockMode.WRITE)) {
      mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
      if (!options.isAlluxioOnly())
        mMountTable.checkUnderWritableMountPoint(path);
      deleteAndJournal(inodePath, options, journalContext);
    }
  }
```

I mean execute the `mMountTable.checkUnderWritableMountPoint(path);` when `options.isAlluxioOnly()` is false.

**Comment by maobaolong** [comment]
@Jason0812 @yupeng9  @uronce-cc In consideration of I have finish this issue in the #5950, i think this PR can be closed.

**Comment by yupeng9** [comment]
sgtm. Thanks @maobaolong

**Pull request closed without merging**